src/drivers/nuttx/lv_nuttx_lcd.c

```diff
@@ -165,28 +165,26 @@ static lv_display_t * lcd_init(int fd, int hor_res, int ver_res)
         return NULL;
     }
 
-    uint32_t px_size = lv_color_format_get_size(lv_display_get_color_format(disp));
+    lv_color_format_t cf = lv_display_get_color_format(disp);   /* Use default cf */
 #if LV_NUTTX_LCD_BUFFER_COUNT > 0
-    uint32_t buf_size = hor_res * ver_res * px_size;
     lv_display_render_mode_t render_mode = LV_DISPLAY_RENDER_MODE_FULL;
 #else
-    uint32_t buf_size = hor_res * LV_NUTTX_LCD_BUFFER_SIZE * px_size;
     lv_display_render_mode_t render_mode = LV_DISPLAY_RENDER_MODE_PARTIAL;
 #endif
 
-    draw_buf = lv_malloc(buf_size);
+    draw_buf = lv_draw_buf_create(hor_res, ver_res, cf, LV_STRIDE_AUTO);
     if(draw_buf == NULL) {
         LV_LOG_ERROR("display draw_buf malloc failed");
```

Comment:
Just a small nit, update the log error message to reflect the new allocation method.